File: evals/tasks/ionwave.ts
import { EvalFunction } from "@/types/evals";

export const ionwave: EvalFunction = async ({
  debugUrl,
  sessionUrl,
  v3,
  logger,
}) => {
  try {
    const page = v3.context.pages()[0];
    await page.goto(
      "https://browserbase.github.io/stagehand-eval-sites/sites/ionwave/",
    );

    await v3.act({ instruction: 'Click on "Closed Bids"' });

    const expectedUrl =
      "https://browserbase.github.io/stagehand-eval-sites/sites/ionwave/closed-bids.html";
    const currentUrl = await page.url();

    return {
      _success: currentUrl.startsWith(expectedUrl),
      currentUrl,
      debugUrl,
      sessionUrl,
      logs: logger.getLogs(),
    };
  } catch (error) {
    return {
      _success: false,
      error: error,
      debugUrl,
      sessionUrl,
      logs: logger.getLogs(),
    };
  } finally {
    await v3.close();
  }
};

File: packages/sdk-ruby/evals/tasks/act/ionwave.rb
# frozen_string_literal: true

# Port of packages/evals/tasks/bench/act/ionwave.ts

Evals.define_task("ionwave") do |t|
  t.page.goto("https://browserbase.github.io/stagehand-eval-sites/sites/ionwave/")

  t.stagehand.act('Click on "Closed Bids"')

  expected_url = "https://browserbase.github.io/stagehand-eval-sites/sites/ionwave/closed-bids.html"
  current_url = t.page.url

  { _success: current_url.start_with?(expected_url), currentUrl: current_url }
end
